_inc/client/writing/index.jsx

```diff
@@ -22,108 +24,137 @@ import {
-	// Put modules available to non-admin user at the top of the list.
-	if ( ! isAdmin ) {
-		let cardsCopy = cards.slice();
-		cardsCopy.reverse().forEach( ( element ) => {
-			if ( includes( nonAdminAvailable, element[0] ) ) {
-				cards.unshift( element );
-			}
-		} );
-		cards = cards.filter( ( element, index ) => cards.indexOf( element ) === index );
-	}
-	cards = cards.map( ( element, i ) => {
-		if ( ! includes( moduleList, element[0] ) ) {
-			return null;
-		}
-		var unavailableInDevMode = props.isUnavailableInDevMode( element[0] ),
-			customClasses = unavailableInDevMode ? 'devmode-disabled' : '',
-			toggle = '',
-			adminAndNonAdmin = isAdmin || includes( nonAdminAvailable, element[0] );
-		if ( unavailableInDevMode ) {
-			toggle = __( 'Unavailable in Dev Mode' );
-		} else if ( isAdmin ) {
-			toggle = <ModuleToggle slug={ element[0] }
-				activated={ isModuleActivated( element[0] ) }
-				toggling={ isTogglingModule( element[0] ) }
-				toggleModule={ toggleModule } />;
-		}
+export const Writing = React.createClass( {
+	displayName: 'WritingSettings',
 
-		if ( 1 === element.length ) {
-			return ( <h1 key={ `section-header-${ i }` /* https://fb.me/react-warning-keys */ } >{ element[0] }</h1> );
-		}
+	propTypes: {
+	},
 
-		return adminAndNonAdmin ? (
-			<FoldableCard
-				className={ customClasses }
-				key={ `module-card_${element[0]}` /* https://fb.me/react-warning-keys */ }
-				header={ element[1] }
-				subheader={ element[2] }
-				summary={ toggle }
-				expandedSummary={ toggle }
-				clickableHeaderText={ true }
-				onOpen={ () => analytics.tracks.recordEvent( 'jetpack_wpa_settings_card_open',
-					{
-						card: element[0],
-						path: props.route.path
-					}
-				) }
-			>
-				{ isModuleActivated( element[0] ) || 'scan' === element[0] ?
-					<AllModuleSettings module={ getModule( element[0] ) } siteAdminUrl={ props.siteAdminUrl } /> :
-					// Render the long_description if module is deactivated
-					<div dangerouslySetInnerHTML={ renderLongDescription( getModule( element[0] ) ) } />
-				}
-				<div className="jp-module-settings__read-more">
-					<Button borderless compact href={ element[3] }><Gridicon icon="help-outline" /><span className="screen-reader-text">{ __( 'Learn More' ) }</span></Button>
-				</div>
-			</FoldableCard>
-		) : false;
-	} );
+	render() {
+		let markdown = this.props.getModule( 'markdown' );
+		let atd = this.props.getModule( 'after-the-deadline' );
+
+		let Composing = moduleSettingsForm(
+			React.createClass( {
+				getCheckbox( setting, label, isAtd = true ) {
+					return(
+						<ModuleSettingCheckbox
+							name={ setting }
+							module={ isAtd ? atd : markdown }
+							label={ label }
+							{ ...this.props }
+						/>
+					);
+				},
 
-	return (
-		<div>
-			{ cards }
-		</div>
-	);
-};
+				render() {
+					return (
+						<form>
+							<SectionHeader label={ __( 'Composing', { context: 'Settings header' } ) }>
+								<Button
```

Comment:
The button also should be clickable when no changes are present (just like in Calypso). Even if nothing has changed.

When clicked, it should change to disabled and the text should change to "Saving...".